core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -200,8 +202,10 @@ public CatalogManager(Config config, EntityStore store, IdGenerator idGenerator)
             .expireAfterAccess(cacheEvictionIntervalInMs, TimeUnit.MILLISECONDS)
             .removalListener(
                 (k, v, c) -> {
-                  LOG.info("Closing catalog {}.", k);
-                  ((CatalogWrapper) v).close();
+                  if (c == RemovalCause.EXPIRED) {
```

Comment:
@jerryshao 
Here, we only handle cache value that was removed by time expiration, so we need to handle manually when using `invalidate`

Reply:
Can you add some comment here?

Reply:
OK

Reply:
Using `EXPIRED` to check may not be so robust, there might be other causes like others (SIZE, COLLECTED) may also need to call the `close`. Also here only when `EXPIRED` you call `close`, you may also leak the `catalog` object since it is not closed.